ts/features/bonus/cgn/saga/orchestration/activation/activationSaga.ts

```diff
@@ -53,6 +54,11 @@ export function* handleCgnStartActivationSaga(): SagaIterator {
   if (result === "completed") {
     if (initialScreen?.name === BONUS_ROUTES.BONUS_AVAILABLE_LIST) {
       yield* call(navigateBack);
+    } else if (initialScreen?.name === WalletRoutes.WALLET_CARD_ONBOARDING) {
+      yield* call(NavigationService.navigate, ROUTES.MAIN, {
+        screen: ROUTES.WALLET_HOME,
+        params: { newMethodAdded: true }
+      });
     }
     yield* call(navigateToCgnDetails);
```

Comment:
if initialScreen is undefined we navigate to CgnDetails?

Reply:
@hevelius yes, `initialScreen` is used only to determine if we need to perform extra navigation actions before display the CGN details screen.